### ISSUE TITLE
Fine-tuning fits calibration temperatures on a slice of the training data

### ISSUE THREAD

**Issue by ashaazami** [NandhaKishorM/laya#186]
Noticed while reading the fine-tuning code. Checked against `laya` 0.3.5 (`573e5b6`) and the
checkpoints on Hugging Face on 2026-09-22.

In the typed-decisions notebook, the post-training temperatures are fitted on a slice of the same
items the run just trained on:

```python
calib_items = all_items[::15][:400]
```

`all_items` is the preprocessed `LocalLLaMA/typed-decisions` **train** split, so the calibration
set is a subset of the training set rather than held-out data.

The fitted values in the published `convaiinnovations/laya-typed-decisions` config are
`[1.0148, 1.0374, 1.0575]`, all within 6% of 1.0, which is what fitting on already-fitted data
tends to produce.

For context, the open PRs that touch this code keep the same behaviour:
[#53](https://github.com/NandhaKishorM/laya/pull/53) moves the loop into `laya/finetune.py` with
`calib_items = items[::calib_every][:calib_max]`, and
[#139](https://github.com/NandhaKishorM/laya/pull/139) changes which temperatures take precedence,
not which data they come from. [#164](https://github.com/NandhaKishorM/laya/pull/164) (draft) adds
separate held-out calibration utilities alongside the notebook path.

Is fitting on the training slice intentional here, or would a held-out split be the intended
behaviour? The dataset ships a separate `test` split, and `train` is large enough to hold a slice
back.

**Comment by Parzifal101** [comment]
Ran into this one too, and opened #191 for it.

One extra measurement, in case it helps make the case. I took `fit_one_temp` straight out of the
notebook and ran it on a linear head trained to convergence on half a synthetic set, fitting the
temperature once on the half it trained on and once on the half it didn't:

| temperature fitted on | value | ECE on unseen data |
|---|---:|---:|
| nothing (uncalibrated) | 1.0000 | 0.1269 |
| the train split | 0.1769 | 0.1692 |
| held-out data | 2.9558 | 0.0580 |

So it's not just that the fit is uninformative — here it comes out worse than leaving the
temperature at 1.0, because the optimiser drives it below 1.0 and sharpens the logits.

Which connects your report to something already in the tree: `clamp_temperature` and its
`RuntimeWarning` exist because the shipped `choice:11+` bucket is `0.1006`, a sharpening value the
runtime now refuses to apply. Same direction, same pathology.

The PR only changes where the calibration items come from — it holds the slice out before
sharding, under a fixed rank-independent seed. It stays off bucket precedence (#139) and the
held-out evaluation tooling (#164), and leaves the published checkpoints' temperatures as they are.
